bleach/linkifier.py

```diff
@@ -59,15 +59,30 @@ def build_url_re(tlds=TLDS, protocols=html5lib_shim.allowed_protocols):
 PROTO_RE = re.compile(r'^[\w-]+:/{0,3}', re.IGNORECASE)
 
 
-EMAIL_RE = re.compile(
-    r"""(?<!//)
-    (([-!#$%&'*+/=?^_`{}|~0-9A-Z]+
-        (\.[-!#$%&'*+/=?^_`{}|~0-9A-Z]+)*  # dot-atom
-    |^"([\001-\010\013\014\016-\037!#-\[\]-\177]
-        |\\[\001-\011\013\014\016-\177])*"  # quoted-string
-    )@(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+[A-Z]{2,6})  # domain
-    """,
-    re.IGNORECASE | re.MULTILINE | re.VERBOSE)
+def build_email_re(tlds=TLDS):
+    """Builds the email regex used by linkifier
+
+   If you want a different set of tlds, pass those in and stomp on the existing ``email_re``::
+
+       from bleach import linkifier
+
+       my_email_re = linkifier.build_email_re(my_tlds_list)
+
+       linker = LinkifyFilter(email_re=my_url_re)
+
+    """
+    return re.compile(
+        r"""(?<!//)
+        (([-!#$%%&'*+/=?^_`{}|~0-9A-Z]+
+            (\.[-!#$%%&'*+/=?^_`{}|~0-9A-Z]+)*  # dot-atom
+        |^"([\001-\010\013\014\016-\037!#-\[\]-\177]
+            |\\[\001-\011\013\014\016-\177])*"  # quoted-string
+        )@(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+(?:%s))  # domain
+        """ % '|'.join(tlds),
```

Comment:
Having to double the other `%` chars is confusing. Can you change that to a `str.format` like `build_url_re`?

Reply:
Since there are ``{}`` inside the regex, this would be just as confusing, wouldn't it?

Reply:
Yeah, good point. We probably need a comment either way. I'd still like to stick with `str.format` for consistency with the `build_url_re`.

Reply:
@g-k Changed it to `str.format` as requested :)

Reply:
Thanks!